workspaces/src/network/block.rs

```diff
@@ -0,0 +1,59 @@
+use near_primitives::views::{BlockHeaderView, BlockView};
+
+use crate::{BlockHeight, CryptoHash};
+
+/// Struct containing information on block coming from the network
+pub struct Block {
+    header: BlockHeader,
+}
+
+impl Block {
+    /// The block timestamp in nanoseconds.
+    pub fn timestamp(&self) -> u64 {
+        self.header.timestamp_nanosec
+    }
+
+    /// Current height of this block.
+    pub fn height(&self) -> BlockHeight {
+        self.header.height
+    }
+
+    /// The hash of the block itself.
+    pub fn hash(&self) -> &CryptoHash {
+        &self.header.hash
+    }
+
+    /// The id of an epoch this block belongs to.
+    pub fn epoch_id(&self) -> &CryptoHash {
+        &self.header.epoch_id
+    }
+}
+
+/// The block header info. This is a non-exhaustive list of items that
+/// could be present in a block header. More can be added in the future.
+#[non_exhaustive]
+struct BlockHeader {
+    height: BlockHeight,
+    epoch_id: CryptoHash,
+    hash: CryptoHash,
+    timestamp_nanosec: u64,
+}
```

Comment:
How did you judge which items to include here, user feedback? Not that I am against it, just wondering

Reply:
ahh I forgot to put this in the initial PR post. The fields I grabbed from the block header was what seemed important for now and was thinking of exposing more later. Would you have any thoughts of including them all now?

Reply:
Nah, that's probably good enough for now. We can always add more, taking stuff out will be more difficult